core/src/main/java/org/apache/druid/initialization/DruidModule.java

```diff
@@ -25,6 +25,13 @@
 import java.util.List;
 
 /**
+ * A Guice module which also provides Jackson modules.
```

Comment:
Text reads funny to me: the second sentence says "you only need this if you are registering Jackson modules" and the next sentence says "well, OK, also you need it for extensions". Could be clearer. Maybe:

> A Guice module that is discoverable by ExtensionsLoader and can optionally provide Jackson modules.
>
> Built-in implementations that do not provide Jackson modules can implement the simpler Module interface instead.

Or, we could add a default impl for getJacksonModules, and then I don't think we need to bother with the guidance. (Since then implementing Module or DruidModule would be equal amounts of code, so there's no real reason to point people at Module.)

Reply:
If the hardest thing in CS is naming, the second hardest is to write comments...

Bingo on the default implementation for `getJacksonModules()`. That simplifies things quite a bit.

The comment was revised accordingly.